exchanges/account/account.go

```diff
@@ -167,26 +169,30 @@ func GetBalance(exch, subAccount string, creds *Credentials, ai asset.Item, c cu
 		return nil, fmt.Errorf("%s %w", exch, ErrExchangeHoldingsNotFound)
 	}
 
-	subAccounts, ok := accounts.SubAccounts[*creds]
+	subAccounts, ok := accounts.subAccounts[*creds]
 	if !ok {
 		return nil, fmt.Errorf("%s %s %w",
 			exch, creds, errNoCredentialBalances)
 	}
 
-	bal, ok := subAccounts[key.SubAccountCurrencyAsset{
+	assets, ok := subAccounts[key.SubAccountAsset{
 		SubAccount: subAccount,
-		Currency:   c.Item,
 		Asset:      ai,
 	}]
 	if !ok {
 		return nil, fmt.Errorf("%s %s %s %s %w",
 			exch, subAccount, ai, c, errNoExchangeSubAccountBalances)
 	}
+	bal, ok := assets.balances[c.Item]
+	if !ok {
+		return nil, fmt.Errorf("%s %s %s %s %w",
+			exch, subAccount, ai, c, errNoExchangeSubAccountBalances)
+	}
```

Comment:
🐒 Nitpick: `ai` is a idiosyncratic var name and if you DriveBy replace it with `a`
🐒 Other errs in this func suffer the same problems, so fixing them all would be nice
* The error is coming at the end seems like it's going to be weird
* We never checked if subAccount is empty. I'd vote we did that earlier around L149. In Lieu of checking it, we need to quote it in errors as "\`%s\`" in case it's empty
* Don't think we need to split to 2 lines, but you tell me how it looks on your screen with the longer message below
So overall I'd suggest:
```suggestion
		return nil, fmt.Errorf("%w for %s SubAccount `%s` currency %s %s %s %w",			errNoExchangeSubAccountBalances, exch, subAccount, a, c)
	}
```